lib/ladybug/energy_model/model.rb

```diff
@@ -39,116 +43,83 @@ module Ladybug
   module EnergyModel
     class Model
       attr_reader :errors, :warnings
-    
+
       # Read Ladybug Energy Model JSON from disk
-      def initialize(file)
+      def self.read_from_disk(file)
+        hash = nil
+        File.open(File.join(file), 'r') do |f|
+          hash = JSON.parse(f.read, symbolize_names: true)
+        end
+
+        Model.new(hash)
+      end
+
+      # Load ModelObject from symbolized hash
+      def initialize(hash)
         # initialize class variable @@extension only once
         @@extension ||= Extension.new
         @@schema ||= @@extension.schema
 
-        @file = file
-        @model = nil
-        File.open(File.join(file), 'r') do |f|
-          @model = JSON::parse(f.read, {symbolize_names: true})
-        end
-        
-        @model_type = @model[:type]
-        raise 'Unknown model type' if @model_type.nil?
-        
-        @face_by_face = false
-        if @model[:type] == 'FaceByFaceModel'
-          @face_by_face = true
-        elsif @model[:type] == 'Model'
-          # no-op
-        else
-          raise "Unknown model type '#{@model[:type]}'"
-        end
+        @hash = hash
+        @type = @hash[:type]
+        raise 'Unknown model type' if @type.nil?
+        raise "Incorrect model type '#{@type}'" unless @type == 'Model'
       end
-      
+
       # check if the model is valid
       def valid?
-        return JSON::Validator.validate(@model, @@schema, {:fragment => "#/components/schemas/#{@model_type}"})
+        JSON::Validator.validate(@hash, @@schema)
       end
-      
+
       # return detailed model validation errors
       def validation_errors
-        return JSON::Validator.fully_validate(@model, @@schema, {:fragment => "#/components/schemas/#{@model_type}"})
-      end
-      
-      # convert to a new openstudio model
-      def to_openstudio
-        osm = OpenStudio::Model::Model.new
-        create_openstudio_objects(osm)
-        return osm
+        JSON::Validator.fully_validate(@hash, @@schema)
```

Comment:
This won't run custom validation checks, that ok?

Reply:
I can add the custom validations in the schema to the ruby code.

Reply:
Well the issue is that we want to get the validation errors from each object, let me think about this one https://github.com/ladybug-tools-in2/energy-model-measure/issues/3